library/classes/rulesets/PQRS/reports/PQRS_0008/Exclusion1.php

```diff
@@ -0,0 +1,42 @@
+<?php
+/**
+ * PQRS Measure 0008 -- Exclusion 
+ *
+ * Copyright (C) 2015 - 2017      Suncoast Connection
+  * 
+ * LICENSE: This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0
+ * See the Mozilla Public License for more details. 
+ * If a copy of the MPL was not distributed with this file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ * 
+ * @author  Art Eaton <art@suncoastconnection.com>
+ * @author  Bryan lee <bryan@suncoastconnection.com>
+ * @package LibreHealthEHR 
+ * @link    http://suncoastconnection.com
+ * @link    http://librehealth.io
+ *
+ * Please support this product by sharing your changes with the LibreHealth.io community.
+ */
+
+class PQRS_0008_Exclusion1 extends PQRSFilter
+{
+    public function getTitle() 
+    {
+        return "Exclusion";
+    }
+    
+    public function test( PQRSPatient $patient, $beginDate, $endDate )
+    {
+$query =
+"SELECT COUNT(b1.code) as count".  
+" FROM billing AS b1".
+" JOIN form_encounter AS fe ON (b1.encounter = fe.encounter)".
+" WHERE b1.pid = ? ".
+" AND fe.date BETWEEN '".$beginDate."' AND '".$endDate."' ".
```

Comment:
right, but $beginDate and $endDate are both user-supplied variables that are validated as a date type (but treated as a string by mysql in $query) while the hard-coded HCPCS codes and the like are not something that the report can pass to the functions.  Nor are all the values in the code tables (like 'code_list_a').  They are all treated as strings.  If you have the ability to put "DROP `Bobby Tables`" into one of those tables, and somehow get it to run here in this SELECT COUNT(), then we have lots of trouble already. 
  If you would, look at measure 447 Initial Patient Population file.  Gimmie a writeup as to how you think it should be done, and I will see about getting it to work out.